repository/src/test/java/org/alfresco/repo/event2/EventGeneratorTest.java

```diff
@@ -0,0 +1,221 @@
+            if (receivedEvents.size() == 1)
+                break;
+        }
+
+        assertFalse("No messages were received!", receivedEvents.isEmpty());
+
+        RepoEvent<?> sent = getRepoEvent(1);
+        RepoEvent<?> received = receivedEvents.get(0);
+
+        assertEquals("Events are different!", sent, received);
+    }
+
+    private static String getText(Message message)
+    {
+        try
+        {
+            ActiveMQTextMessage am = (ActiveMQTextMessage) message;
+            return am.getText();
+        }
+        catch (JMSException e)
+        {
+            return null;
+        }
+    }
+
+    // a simple main to investigate he contents of the local broker
+    public static void main(String[] args) throws Exception
+    {
+        dumpBroker("tcp://localhost:61616");
+        System.exit(0);
+    }
+
+    private static void dumpBroker(String url) throws Exception
+    {
+        System.out.println("Broker at url: '" + url + "'");
+
+        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(url);
+        ActiveMQConnection connection = (ActiveMQConnection) connectionFactory.createConnection();
+        try
+        {
+            connection.start();
+
+            DestinationSource ds = connection.getDestinationSource();
+
+            Set<ActiveMQQueue> queues = ds.getQueues();
+            System.out.println("\nFound " + queues.size() + " queues:");
+            for (ActiveMQQueue queue : queues)
+            {
+                try
+                {
+                    System.out.println("- " + queue.getQueueName());
+                }
+                catch (JMSException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+
+            Set<ActiveMQTopic> topics = ds.getTopics();
+            System.out.println("\nFound " + topics.size() + " topics:");
+            for (ActiveMQTopic topic : topics)
+            {
+                try
+                {
+                    System.out.println("- " + topic.getTopicName());
+                }
+                catch (JMSException e)
+                {
+                    e.printStackTrace();
+                }
+            }
+            
+            Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            Destination destination = session.createTopic(EVENT2_TOPIC_NAME);
+            MessageConsumer consumer = session.createConsumer(destination);
+
+            System.out.println("\nListening to topic "+EVENT2_TOPIC_NAME+"...");
+            consumer.setMessageListener(new MessageListener()
+            {
+                @Override
+                public void onMessage(Message message)
+                {
+                    String text = getText(message);
+                    System.out.println("Received message " + message + "\n" + text+"\n");
+                }
+            });
+            
+            Thread.sleep(50000000l);
```

Comment:
Can you please make this sleep configurable and passed through the "main" as a parameter?

Reply:
Sure, I will do that.